pkg/pipeline/extract/conntrack/conntrack.go

```diff
@@ -228,7 +213,7 @@ func NewConnectionTrack(opMetrics *operational.Metrics, params config.StageParam
 	metrics := newMetrics(opMetrics)
 	conntrack := &conntrackImpl{
 		clock:                        clock,
-		connStore:                    newConnectionStore(metrics),
+		connStore:                    newConnectionStore(cfg.Scheduling, metrics, clock.Now),
```

Comment:
Why is clock passed as a parameter to NewConnectionTrack() rather than defining it here?

Reply:
`NewConnectionTrack()` accepts the clock as an argument to allow mocking it in the unit tests.
Mocking example:
https://github.com/netobserv/flowlogs-pipeline/blob/67fb882dab394b808ab09716974ba55145497340/pkg/pipeline/extract/conntrack/conntrack_test.go#L608
https://github.com/netobserv/flowlogs-pipeline/blob/67fb882dab394b808ab09716974ba55145497340/pkg/pipeline/extract/conntrack/conntrack_test.go#L613
https://github.com/netobserv/flowlogs-pipeline/blob/67fb882dab394b808ab09716974ba55145497340/pkg/pipeline/extract/conntrack/conntrack_test.go#L657